iot-e2e-tests/android/things/src/androidTest/java/com/microsoft/azure/sdk/iot/androidthings/iothubservices/errorinjection/messaging/SendMessagesErrInjModuleThingsRunner.java

```diff
@@ -0,0 +1,69 @@
+/*
+ *  Copyright (c) Microsoft. All rights reserved.
+ *  Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ */
+
+package com.microsoft.azure.sdk.iot.androidthings.iothubservices.errorinjection.messaging;
+
+import com.microsoft.azure.sdk.iot.androidthings.BuildConfig;
+import com.microsoft.azure.sdk.iot.androidthings.helper.TestGroupA;
+import com.microsoft.azure.sdk.iot.common.helpers.ClientType;
+import com.microsoft.azure.sdk.iot.common.helpers.Rerun;
+import com.microsoft.azure.sdk.iot.common.tests.iothubservices.errorinjection.SendMessagesErrInjTests;
+import com.microsoft.azure.sdk.iot.deps.util.Base64;
+import com.microsoft.azure.sdk.iot.device.InternalClient;
+import com.microsoft.azure.sdk.iot.device.IotHubClientProtocol;
+import com.microsoft.azure.sdk.iot.device.exceptions.ModuleClientException;
+import com.microsoft.azure.sdk.iot.service.BaseDevice;
+import com.microsoft.azure.sdk.iot.service.auth.AuthenticationType;
+import com.microsoft.azure.sdk.iot.service.exceptions.IotHubException;
+
+import org.junit.AfterClass;
+import org.junit.Rule;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.security.GeneralSecurityException;
+import java.util.Collection;
+
+@TestGroupA
+@RunWith(Parameterized.class)
+public class SendMessagesErrInjModuleThingsRunner extends SendMessagesErrInjTests
+{
+    static Collection<BaseDevice> identities;
+
+    @Rule
+    public Rerun count = new Rerun(3);
+
+    public SendMessagesErrInjModuleThingsRunner(InternalClient client, IotHubClientProtocol protocol, BaseDevice identity, AuthenticationType authenticationType, String clientType, String publicKeyCert, String privateKey, String x509Thumbprint)
+    {
+        super(client, protocol, identity, authenticationType, clientType, publicKeyCert, privateKey, x509Thumbprint);
+    }
+
+    //This function is run before even the @BeforeClass annotation, so it is used as the @BeforeClass method
+    @Parameterized.Parameters(name = "{1}_{3}_{4}")
+    public static Collection inputs() throws IOException, IotHubException, GeneralSecurityException, URISyntaxException, ModuleClientException, InterruptedException
+    {
+        String privateKeyBase64Encoded = BuildConfig.IotHubPrivateKeyBase64;
+        String publicKeyCertBase64Encoded = BuildConfig.IotHubPublicCertBase64;
+        iotHubConnectionString = BuildConfig.IotHubConnectionString;
+        String x509Thumbprint = BuildConfig.IotHubThumbprint;
+        String privateKey = new String(Base64.decodeBase64Local(privateKeyBase64Encoded.getBytes()));
+        String publicKeyCert = new String(Base64.decodeBase64Local(publicKeyCertBase64Encoded.getBytes()));
+
+        Collection inputs = inputsCommon(ClientType.MODULE_CLIENT, publicKeyCert, privateKey, x509Thumbprint);
+
```

Comment:
Is it possible to build these strings from a common place? I notice these are used in every class runner. It would be easy and clean if they can be modified from a single place (when it happens).

Reply:
Discussed with Tim offline and we both agreed that leaving these at the current form (one liner getting values from buildConfig) is a better choice for ease of debugging and readibility.  Moreover, there is a task to move to using the cert generator, by then, these sections would be cleaned up.